src/Stratis.Bitcoin.Features.Miner/PosMinting.cs

```diff
@@ -164,45 +173,54 @@ public class CoinstakeContext
         /// but high enough to compensate for tasks' overhead.</remarks>
         private const int UtxoStakeDescriptionsPerCoinstakeWorker = 25;
 
+        /// <summary>Responsible for downloading and validating blocks.</summary>
         private readonly ConsensusLoop consensusLoop;
+        /// <summary>Chain of headers from genesis.</summary>
         private readonly ConcurrentChain chain;
-
         /// <summary>Specification of the network the node runs on - regtest/testnet/mainnet.</summary>
         private readonly Network network;
+        /// <summary>Used to verify that node is connected to network before we start staking.</summary>
         private readonly IConnectionManager connection;
+        /// <summary>Used to verify that node is not in a state of IBD (Initial Block Download).</summary>
+        private readonly ChainState chainState;
+        /// <summary>Provides date time functionality.</summary>
         private readonly IDateTimeProvider dateTimeProvider;
+        /// <summary>Used for creating block template.</summary>
         private readonly AssemblerFactory blockAssemblerFactory;
-        private readonly IBlockRepository blockRepository;
-        private readonly ChainState chainState;
-        private readonly Signals.Signals signals;
+        /// <summary>Allows consumers to perform cleanup during a graceful shutdown.</summary>
         private readonly INodeLifetime nodeLifetime;
-        private readonly NodeSettings settings;
+        /// <summary>Used to fetch UTXOs.</summary>
         private readonly CoinView coinView;
-
         /// <summary>Database of stake related data for the current blockchain.</summary>
         private readonly StakeChain stakeChain;
+        /// <summary>Used for creating <see cref="mining"/> loop.</summary>
         private readonly IAsyncLoopFactory asyncLoopFactory;
+        /// <summary>A manager providing operations on wallets.</summary>
         private readonly WalletManager walletManager;
+        /// <summary>Provides value for PoS reward and checks PoS kernel.</summary>
         private readonly PosConsensusValidator posConsensusValidator;
-
         /// <summary>Factory for creating loggers.</summary>
         private readonly ILoggerFactory loggerFactory;
-
         /// <summary>Instance logger.</summary>
         private readonly ILogger logger;
-
+        /// <summary>Mining loop.</summary>
         private IAsyncLoop mining;
+        /// <summary>Balance that shouldn't participate in taking.</summary>
         private Money reserveBalance;
+        /// <summary>UTXO's value threshold for being selected for staking.</summary>
         private readonly int minimumInputValue;
+        /// <summary>Time in milliseconds between attempts to generate PoS blocks.</summary>
         private readonly int minerSleep;
 
+        /// <summary>Used for asynchronously calling <see cref="mempool"/>.</summary>
```

Comment:
that does not make any sense actually
lock description should describe lock contract 
and the purpose of locks is not to async call something
the purpose of the lock is to have an exclusive access or in case of RW lock, to prevent readers and writers to operate at the same time